src/app/gateways/views/GatewayListTabsView.vue

```diff
@@ -48,3 +48,9 @@
 <script lang="ts" setup>
 import LinkBox from '@/app/common/LinkBox.vue'
 </script>
+<style lang="scss" scoped>
+:deep(.link-box) {
+  margin-top: -10px;
+  margin-bottom: -8px;
+}
```

Comment:
**Issue**: We should really, really not use styles like this for general alignment matters. There is no reason the vertical margins of the targeted element here should be pulled in like that. This should be removed everywhere where it was added in this PR.

Reply:
I mention this in the desc:

> There are few semi-hacks here such as minus margins, but its to keep the certain buttons inline whilst keeping margins the same whether the button appears or not, and is probably the best way to do this (hence only a semi-hack)

Happy to use another approach to keep the margins the same whether there is a button or not. Previous to this PR the code that was there meant that the margins were out here:

![Screenshot 2024-03-25 at 09 14 03](https://github.com/kumahq/kuma-gui/assets/554604/5362d4b4-3553-4172-b3e6-429ec250a47b)

Let me know if you have a different approach that works.

Reply:
Oh this specific one was for when we show the linkbox and when we don't. Because our headers are now much smaller, having a linkbox conditionally show means that the margins can look wrong when we have a linkbox.

With this change in either case now the margins for the header always the same.

Reply:
Using negative margins to artificially reduce the height of the button in the actions element just so that the title bar is the same height as if the button wasn’t there is a pretty bad approach to make the title bars the same height:

- It’s not accurate. In this PR, the title bar (i.e. the `.card-header` element) without button has a height of 21 pixels. The title bar with a button has a height of 21.9 pixels. So they’re not actually aligned.
- It looks visually bad. The button now clips into the KCard padding which it really shouldn’t.
- It’s brittle because it depends on the height of the buttons. If the button height changes, the offset values will need to be adjusted.

There are two other strategies to handle this:

- Fixed height. Making the title bar the same height everywhere so that horizontally laid out elements like on our overview page align with their content. Easy to implement. The only downside is that it, too, might need to be manually adjusted if the height of the action items or headings changes. But at least it only needs to be done in one place. Strictly better than the negative margin approach.
- Not to bother. Instead, let content be laid naturally. Even easier to "implement" (it needs no implementation, just code removal). The content in the overview page wouldn’t be aligned, but that’s alright in my opinion.

Reply:
> It’s not accurate.

Are you saying its 0.1 pixel out?

> It looks visually bad.

Could you screengrab what you are seeing?

> It’s brittle because it depends on the height of the buttons. If the button height changes, the offset values will need to be adjusted.

Agreed, but thats not a huge issue I don't think.

> There are two other strategies to handle this:

Both of these have their problems, and most notably the second one doesn't solve the issue. We want these things to be aligned and to be consistent with all our other headers.

If you could pop up a screengrab of what you are looking at that looks visually bad it would help (also to confirm exactly which one of these instances we are talking about) that'd be great! ty

Reply:
> Are you saying its 0.1 pixel out?

0.9 pixels.

> Could you screengrab what you are seeing?

![image](https://github.com/kumahq/kuma-gui/assets/5774638/8f59e4f7-144a-4604-86ad-3ceea6458960)

The top padding is too short.

Reply:
Looks spot on to me?

![Screenshot 2024-03-25 at 12 22 47](https://github.com/kumahq/kuma-gui/assets/554604/bfad2daa-d475-4fc4-a3fd-0e4a3d54637c)

Which is the bit that looks visually bad?

Reply:
I think everything should line up.

I think this stuff you are talking about "clipping the padding" is quite a common thing in user interfaces, and generally an approach to doing that is to use negative margins.

I think this looks more than fine, and better than it did previously 🤷

Reply:
Heres another example of "clipping the padding":

![Screenshot 2024-03-25 at 12 34 01](https://github.com/kumahq/kuma-gui/assets/554604/b4412d7f-85bd-46c2-9bf1-9c9d2e381daa)

Reply:
What about this:

![Screenshot 2024-03-25 at 12 53 28](https://github.com/kumahq/kuma-gui/assets/554604/3585cacc-7618-4bf1-80ec-08697d118133)

Reply:
What about it? That UI doesn’t seem related to what we’re discussing here.

Reply:
Well its our UI and the baselines of the text and the button text line up?

Reply:
Here's how it was previously, note how the padding, especially for the Mesh title, looks far too big because of the fixed height.

![Screenshot 2024-03-25 at 13 03 22](https://github.com/kumahq/kuma-gui/assets/554604/f2ebc936-5132-4eba-8924-e236bee669a8)

Reply:
> That text happens to line up because (1) the content is vertically centered

Yes, but it "clips the padding" of the box. So its exactly the same as what we are talking about here and its in our UI (which was one of your points)

Reply:
I decided to remove the negative margins here. 

The headers have different margins now, and despite that fact that this PR was supposed to be resolving this I kinda want to get this in 🤷 

![Screenshot 2024-03-26 at 11 53 38](https://github.com/kumahq/kuma-gui/assets/554604/1071cec2-3f96-4466-8c17-ce4d952dab55)